src/cascadia/TerminalSettingsEditor/AISettings.cpp

```diff
@@ -67,28 +75,59 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
             TelemetryPrivacyDataTag(PDT_ProductAndServiceUsage));
     }
 
-    void AISettings::ClearKeyAndEndpoint_Click(const IInspectable& /*sender*/, const RoutedEventArgs& /*e*/)
+    void AISettings::ClearAzureOpenAIKeyAndEndpoint_Click(const IInspectable& /*sender*/, const RoutedEventArgs& /*e*/)
     {
-        _ViewModel.AIEndpoint(L"");
-        _ViewModel.AIKey(L"");
+        _ViewModel.AzureOpenAIEndpoint(L"");
+        _ViewModel.AzureOpenAIKey(L"");
     }
 
-    void AISettings::StoreKeyAndEndpoint_Click(const IInspectable& /*sender*/, const RoutedEventArgs& /*e*/)
+    void AISettings::StoreAzureOpenAIKeyAndEndpoint_Click(const IInspectable& /*sender*/, const RoutedEventArgs& /*e*/)
     {
         // only store anything if both fields are filled
-        if (!EndpointInputBox().Text().empty() && !KeyInputBox().Password().empty())
+        if (!AzureOpenAIEndpointInputBox().Text().empty() && !AzureOpenAIKeyInputBox().Password().empty())
         {
-            _ViewModel.AIEndpoint(EndpointInputBox().Text());
-            _ViewModel.AIKey(KeyInputBox().Password());
-            EndpointInputBox().Text(L"");
-            KeyInputBox().Password(L"");
+            _ViewModel.AzureOpenAIEndpoint(AzureOpenAIEndpointInputBox().Text());
+            _ViewModel.AzureOpenAIKey(AzureOpenAIKeyInputBox().Password());
+            AzureOpenAIEndpointInputBox().Text(L"");
+            AzureOpenAIKeyInputBox().Password(L"");
 
             TraceLoggingWrite(
                 g_hSettingsEditorProvider,
-                "AIEndpointAndKeySaved",
-                TraceLoggingDescription("Event emitted when the user stores an AI key and endpoint"),
+                "AzureOpenAIEndpointAndKeySaved",
+                TraceLoggingDescription("Event emitted when the user stores an Azure OpenAI key and endpoint"),
                 TraceLoggingKeyword(MICROSOFT_KEYWORD_CRITICAL_DATA),
                 TelemetryPrivacyDataTag(PDT_ProductAndServiceUsage));
         }
     }
+
+    void AISettings::ClearOpenAIKey_Click(const IInspectable& /*sender*/, const RoutedEventArgs& /*e*/)
+    {
+        _ViewModel.OpenAIKey(L"");
+    }
+
+    void AISettings::StoreOpenAIKey_Click(const IInspectable& /*sender*/, const RoutedEventArgs& /*e*/)
+    {
+        if (!OpenAIKeyInputBox().Password().empty())
+        {
+            _ViewModel.OpenAIKey(OpenAIKeyInputBox().Password());
+            OpenAIKeyInputBox().Password(L"");

```

Comment:
nit: Avoid calling getters repeatedly and instead assign them to variables. Aids in debugging and avoids unknown side-effects if the getter is non-trivial (e.g. computes the return value, or worse, fetches it from a database, etc.).